manipulation/src/main/java/org/jboss/gm/manipulation/actions/UploadTaskTransformerAction.java

```diff
@@ -1,40 +1,96 @@
 package org.jboss.gm.manipulation.actions;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
 import org.gradle.api.Action;
 import org.gradle.api.Project;
-import org.gradle.api.artifacts.maven.MavenResolver;
-import org.gradle.api.tasks.Upload;
+import org.gradle.api.Task;
+import org.gradle.api.XmlProvider;
+import org.gradle.api.artifacts.ArtifactRepositoryContainer;
+import org.gradle.api.artifacts.repositories.ArtifactRepository;
+import org.gradle.api.logging.Logger;
+import org.gradle.util.GradleVersion;
+import org.jboss.gm.common.logging.GMLogger;
 import org.jboss.gm.common.model.ManipulationModel;
 import org.jboss.gm.manipulation.ResolvedDependenciesRepository;
 
 import static org.jboss.gm.manipulation.ManipulationPlugin.LEGACY_MAVEN_PLUGIN;
 
 /**
- * Fixes pom.xml generation in old "maven" plugin.
+ * Fixes {@code pom.xml} generation in the old &quot;maven&quot; plugin.
+ * <p>
+ * Adds {@code PomTransformer} to all {@code MavenResolver} repositories in {@code Upload} tasks.
  * <p>
- * Adds PomTransformer to all MavenResolver repositories in Upload tasks.
+ * <em>
+ * Gradle 7+ removed support for the old &quot;maven&quot; plugin. This class uses reflection in order to build and run
+ * on Gradle 7+. Due to the removal of the old &quot;maven&quot; plugin, this action does nothing when run on Gradle 7+.
+ * </em>
+ * </p>
  */
 public class UploadTaskTransformerAction implements Action<Project> {
 
-    private final ManipulationModel alignmentConfiguration;
-    private final ResolvedDependenciesRepository resolvedDependenciesRepository;
+    private static final Logger LOGGER = GMLogger.getLogger(UploadTaskTransformerAction.class);
+    private final boolean disabled;
+    private ManipulationModel alignmentConfiguration;
+    private ResolvedDependenciesRepository resolvedDependenciesRepository;
+    private Class<? extends ArtifactRepository> resolverClass;
+    private Method getPomMethod;
+    private Class<? extends Task> uploadClass;
+    private Method getRepositoriesMethod;
+    private Method withXmlMethod;
 
     public UploadTaskTransformerAction(ManipulationModel alignmentConfiguration,
             ResolvedDependenciesRepository resolvedDependenciesRepository) {
+        disabled = GradleVersion.current().compareTo(GradleVersion.version("7.0")) >= 0;
+
+        if (disabled) {
+            return;
+        }
```

Comment:
Out of curiosity, if a build does configure a UploadTask action in Gradle 7, can it ever even get to this point?

Reply:
It was more a guard for the Java code itself as I am not entirely sure when or how the class can be created.

Generally, in a build file itself, it would fail as soon as an attempt is made to use the maven plugin, before the alignment can even be run.

Also, looking at <https://github.com/project-ncl/gradle-manipulator/blob/4fa608d33161af3d24b354fd1a2948518184ee66/manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java#L156>, why wouldn't we move the check from <https://github.com/project-ncl/gradle-manipulator/blob/4fa608d33161af3d24b354fd1a2948518184ee66/manipulation/src/main/java/org/jboss/gm/manipulation/actions/UploadTaskTransformerAction.java#L30>to there.

Can `execute` be called from somewhere else?

In other words, more generally, can we remove any guards from `UploadTaskTransformerAction` and move it to `ManipulationPlugin`? Or is there some other possible way to call this?

Reply:
Yeah I agree, I think the original guards were paranoia but whether we _really_ need them is another question. I don't think it should be possible to call those classes ... as per my comment line in one of the classes `This should never happen due to prior checks in ManipulationPlugin`

Reply:
From looking at https://github.com/bmuschko/gradle-nexus-plugin (now archived) I think the different Upload if check is fine.

While I might be ok with removing some of the checks I would rather this was done in a separate PR.

Given I don't think its possible to hit this `disabled` check in Gradle 7 can we remove it please.